### ISSUE TITLE
TypeScript + JSX (TSX): required trailing commas are stripped from type parameters

### ISSUE THREAD

**Issue by OliverJAsh** [prettier/prettier#6302]
**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzKmC8mAPACoA0AfABQxKYkCUBFmMA3CGSBAA4wCW6ZKACGAJzEQA7gAVxCNMhAiAbhH4ATTiABGYkWADWcGAGUeB-lADmyGGICucLgAsYAWwA2AdRf94aBZgcKYK-vwq-gCeSmBoilxWaHBiMDL61u4iyNginslcAFZoAB4AQvpGJqYi7nAAMlZwOXkFIMUlplbWnnAAig4Q8C35ziAWYsliSjBRPHBoYGL8fNo8y7DemjAuyAAcAAxc6xDJ3vo8SusLKSrNXACOg-DpvIooImgAtFBwcBr-bRiOBPfjA9IiTLZJC5UZcZLufh2RxjNDdXoDIbNGGtMYwEQ6LYaHbIABMXHsIn4nm6AGEIO4skooNB7iAHMkSAT3rDkgBfPlAA)

**Input:**
```tsx
const fn = <T,>(t: T) => t;
```

**Output:**
This is invalid syntax for a `.tsx` file. TypeScript needs the comma because without it, the syntax is ambiguous.
```tsx
const fn = <T>(t: T) => t;
```

- https://github.com/Microsoft/TypeScript/issues/15713#issuecomment-499474386
- https://stackoverflow.com/questions/32308370/what-is-the-syntax-for-typescript-arrow-functions-with-generics#comment99104831_45576880

**Expected behavior:**
```tsx
const fn = <T,>(t: T) => t;
```

**Comment by sosukesuzuki** [comment]
this issue is duplicated to #6114 . Also, fixed in #6115 and #6190 .
But, it behavior depends on file extension(`.tsx`), so we cannot see it in Playground (Playground is not supported file extension).

If you run prettier in CLI and file extension is `tsx`, syntax error does not happen.

**Comment by OliverJAsh** [comment]
Thanks!
